Source/Request.swift

```diff
@@ -405,10 +405,42 @@ open class DownloadRequest: Request {
 
     // MARK: Helper Types
 
-    /// A closure executed once a request has successfully completed in order to determine where to move the temporary
-    /// file written to during the download process. The closure takes two arguments: the temporary file URL and the URL
-    /// response, and returns a single argument: the file URL where the temporary file should be moved.
-    public typealias DownloadFileDestination = (URL, HTTPURLResponse) -> URL
+    /// A collection of options to be executed prior to moving a downloaded file from the temporary URL to the
+    /// destination URL.
+    public struct DownloadOptions: OptionSet {
+        /// Defines the `RawValue` type as `UInt` to satisfy the `RawRepresentable` protocol.
+        public typealias RawValue = UInt
+
+        /// Returns the raw bitmask value of the option and satisfies the `RawRepresentable` protocol.
+        public let rawValue: RawValue
```

Comment:
The private static properties are merely to group the raw values together to make it virtually impossible to screw up adding a new option. I can remove it if you think it's overboard. I'll definitely remove the typealias declaration!